### PR TITLE
Build warning fixes

### PR DESCRIPTION
**Pull request by adutra**
Fixes a few build warnings. There is no functional change, and each commit can be reviewed separately.

### CODE REVIEW

**Review by sfc-gh-mcollado — COMMENTED** [review]
These are great! Just the one question about the constant

**Comment by snazy** [comment]
(Reopening, was closed due to GH's "fix comment" confusion)

File: polaris-core/src/main/java/io/polaris/core/context/CallContext.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ public interface CallContext extends AutoCloseable {
 
   // For requests that make use of a Catalog instance, this holds the instance that was
   // created, scoped to the current call context.
-  public static final String REQUEST_PATH_CATALOG_INSTANCE_KEY = "REQUEST_PATH_CATALOG_INSTANCE";
+  String REQUEST_PATH_CATALOG_INSTANCE_KEY = "REQUEST_PATH_CATALOG_INSTANCE";
```
[the remark this answers is not in the document]
Comment (adutra) [reply polaris-core/src/main/java/io/polaris/core/context/CallContext.java:43]: It still is: this is an interface so `public static final` is implicit.

**Review by snazy — APPROVED** [review]

File: polaris-core/src/main/java/io/polaris/core/context/CallContext.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (collado-mike) [reply polaris-core/src/main/java/io/polaris/core/context/CallContext.java:43]: Dur! 🤦🏽‍♂️

**Pull request merged**